tests/integration/TestKafkaProducer.cc

```diff
@@ -163,7 +166,7 @@ TEST(KafkaSyncProducer, InSyncBrokersAckTimeout)
             catch (const KafkaException& e)
             {
                 std::cout << "[" << Utility::getCurrentTime() << "] Exception caught: " << e.what() << std::endl;
-                EXPECT_EQ(RD_KAFKA_RESP_ERR_REQUEST_TIMED_OUT, e.error().value());
+                EXPECT_EQ(RD_KAFKA_RESP_ERR__MSG_TIMED_OUT, e.error().value());
```

Comment:
@edenhill Another behavior change in librdkafka `v1.5.2`. This test case is using a really short `request.timeout.ms`  (only `1ms`, which should be enforced by the brokers), while the error would be `MSG_TIMED_OUT`, instead of `REQUEST_TIMED_OUT`. (Note: here the `message.timeout.ms` is `1000ms`)

Reply:
Same reason as above, retries is not exhausted but the timeout is.